tsdate/prior.py

```diff
@@ -970,10 +973,22 @@ def fill_priors(node_parameters, timepoints, ts, Ne, *, prior_distr, progress=Fa
     datable_nodes = np.ones(ts.num_nodes, dtype=bool)
     datable_nodes[ts.samples()] = False
     datable_nodes = np.where(datable_nodes)[0]
+
+    """
+    Convert piecewise constant demographic history to coalescent units,
+    then convert coalescent time grid to generations
+    """
+    _, epoch_start, coalescent_rate = util.change_time_measure(
+        np.array([0]), population_size[:, 0], 2 * population_size[:, 1]
+    )
+    generations, _, _ = util.change_time_measure(
+        timepoints, epoch_start, coalescent_rate
+    )
+
     prior_times = base.NodeGridValues(
         ts.num_nodes,
         datable_nodes[np.argsort(ts.tables.nodes.time[datable_nodes])].astype(np.int32),
-        timepoints,
+        generations,
```

Comment:
Ultimately we might want to allow times to be specified in years rather than generations (this makes more sense for e.g. cases with historical samples dated to a specific timepoint), and of course is the only natural unit to use when inferring dates from population with overlapping generations, or where average generation time differs between subpopulations. Previously this would have be done by simply submitting rescaled values (e.g. for mutation rate, Ne, etc) to the prior, and changing the `time_units` parameter. I'm a little worried that hard-coding "generations" as the name here fixes us to a generational timescale?

Perhaps all we need, however, is a comment that mentions a "todo" to allow for different timescales to be used.